### PR TITLE
feat: support webp format with jpg fallback

### PR DESCRIPTION
**Pull request by midudev**
In order to improve web performance we're using webp when available.

- [x] Added support to webp
- [x] Keep `jpg` fallback.

https://user-images.githubusercontent.com/1561955/109354468-2791a180-787e-11eb-8fd0-a4f2bb87689c.mp4

This is addressing some issues from: https://github.com/paulirish/lite-youtube-embed/issues/70

### CODE REVIEW

File: src/lite-yt-embed.js
Comment (laukstein) [inline src/lite-yt-embed.js:96]: There are better techniques on checking webp support like
```
document.createElement("canvas").toDataURL("image/webp").indexOf("data:image/webp") === 0
```

the example above will actually add new network request and will fail if not whitelisted in CSP `img-src data:`.
cc @paulirish

Comment (midudev) [reply src/lite-yt-embed.js:96]: Before adding the one from the PR, I tried that one you're mentioning.

Unfortunately, seems to be problematic with Safari and Firefox.

Anyway, I'm more than happy to change it if the solution based on canvas is more suitable according to @paulirish 👍

Comment (laukstein) [reply src/lite-yt-embed.js:96]: You could simply use html5 approach

```
<picture>
    <source srcset=poster.avif type=image/avif>
    <source srcset=poster.webp type=image/webp>
    <img src=poster.jpg>
</picture>
```

with it can cover more formats, also AVIF.

**Review by paulirish — CHANGES_REQUESTED** [review]
> You could simply use html5 approach
> 
> ```
> <picture>
>     <source srcset=poster.avif type=image/avif>
>     <source srcset=poster.webp type=image/webp>
>     
> </picture>
> ```

yeah i think this is my favored solution... 

this, of course, means switching from a bg-image to using picture/source/srcset, but I suspect that may improve some other aspects.   if anyone wants to give this a shot, i'd appreciate it.

File: src/lite-yt-embed.js
Comment (midudev) [reply src/lite-yt-embed.js:96]: @laukstein 

```
document.createElement("canvas")
          .toDataURL("image/webp")
          .indexOf("data:image/webp") === 0
```

This is returning false on Safari while it supports `webp`. It seems that is not a good solution. :/

And still with the HTML5 approach we would need a way to preload the image before.

**Comment by midudev** [comment]
> > You could simply use html5 approach
> > ```
> > <picture>
> >     <source srcset=poster.avif type=image/avif>
> >     <source srcset=poster.webp type=image/webp>
> >     
> > </picture>
> > ```
> 
> yeah i think this is my favored solution...
> 
> this, of course, means switching from a bg-image to using picture/source/srcset, but I suspect that may improve some other aspects. if anyone wants to give this a shot, i'd appreciate it.

@paulirish I've it already implemented and working with the HTML approach BUT...

We still need a way to detect if we have webp support in order to `preload` the image.
We have two options here:
  a) We remove completely the `preload` functionality as we're already improving moving to webp. 
  
  Good:
    - Less code.
    - It relies on HTML and no JavaScript and weird checks.
  Bad:
    - No preload might have some penaly perf?
  
  b) We implement the "officialish" webp check support from webp docs: https://developers.google.com/speed/webp/faq#how_can_i_detect_browser_support_for_webp

  Good:
    - Preload
  Bad:
    - More code
    - All webp checks could have some pitfalls now or in the future.

Your call @paulirish

**Comment by midudev** [comment]
@paulirish I've it already implemented and working with the HTML approach BUT...

We still could need a way to detect if we have webp support in order to preload the image.

As I see it, we have **three options** here:

a) We remove completely the preload functionality as we're already improving moving to webp.

👍 Less code and it relies on HTML and no JavaScript and weird checks.
👎 No preload might have some penaly perf?

b) We implement the "officialish" webp check support from webp docs: https://developers.google.com/speed/webp/faq#how_can_i_detect_browser_support_for_webp

👍 Preload
👎 More code, runtime aand all webp checks could have some pitfalls now or in the future.

c) We only preload `image/webp` in case it's supported by the browser.

👍  Preload and supports 90% of the users with almost no code.
👎  No preload for 10% of the users with old browsers.

Your call @paulirish.

PS: About the first check of webp support by @laukstein is only working for Chrome.
Safari and Firefox are returning false when they support webp, so it's not reliable.

**Comment by Garbee** [comment]
Personally I'd prefer the option that wastes no bandwidth in any case. C, preloading regardless of support, is wasteful for those who don't have support. This may not be a big concern for most, but in places where data usage does cost it is a massive problem. Every byte matters.

I'd default to option A for now, let the browser make the best choice. Then we can analyze responsible preloading solutions on it's own merit separately.

**Comment by midudev** [comment]
Hey @paulirish. Just ping, waiting for your feedback. 🙇

**Pull request closed without merging**
